### ISSUE TITLE
Editing the name of a person prevents him from being listed as part of the group

### ISSUE THREAD

**Issue by seah-minlong** [AY2425S1-CS2103T-T16-3/tp#200]
To recreate:
1. `createGroup g/SMU m/ 1 2` with persons `Alex Yeoh` and `Bernice Yu`
2. `edit 1 n/Zebra`  -> the whole list will be shown
3.  `search g/SMU` will only list 1 person now
4. `search n/Zebra`
5. `edit 1 n/Alex Yeoh`
6. `search g/SMU` will list 2 persons correctly again

Furthermore, after step 4, if instead of doing step 5.
1. `addToGroup g/SMU m/1` (To add Zebra to SMU)
2. `search g/SMU` will display both lists
3. If you edit Zebra back to Alex Yeoh again, 
   - The list will show both Alex Yeoh and Bernice Yu
   - Now Zebra and Alex Yeoh names, for that `Person`, will be accepted as part of the group
   - However, editing other `Person`s to have either of those names will not lead to them appearing in the group

**Comment by seah-minlong** [comment]
TLDR if you change the name of a person in a group, he will be removed. You change his name back, he will be placed back. Strangely, it seems that the name is tied quite strongly with the Group

**Comment by seah-minlong** [comment]
Editing other fields seem to remove them from the group as well.

This also has a strange interaction with `addToGroup`. While they are removed from the group list view, the message from `addToGroup` says that:
The following users were added to the group SMU:
The following users were already in the group SMU and will remain in the group, the rest of the users have been added accordingly: 
Alex Yeoh
Bernice Yu
```

**Comment by chongsun2002** [comment]
We completely prevent this from happening in commit 9e0c1d4.
